tests/test_cli_compile.py

```diff
@@ -2494,9 +2497,185 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
     assert exp in out.stderr
 
 
+def _parse_deps(text: str, keys: Iterable[str]) -> dict[str, str | None]:
+    """Parse ``pip-compile`` output and return mapping from package name to package version
+
+    :param text: output from ``pip-compile``
+    :param keys: names of expected packages. If not in text the version will be set to ``None``.
+    """
+    result: dict[str, str | None] = {k: None for k in keys}
+
+    for line in text.splitlines():
+        prefix = line.split("#")[0].strip()
+        if not prefix:
+            continue
+        try:
+            k, v = prefix.split("==")
+        except ValueError:
+            continue
+        result[k] = v
+
+    return result
+
+
+class Version:
+    def __init__(self, expected: str) -> None:
+        self._pat = expected
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, str):
+            return False
+
+        return fnmatch.fnmatch(other, self._pat)
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}@{id(self)}(_pat={self._pat!r})>"
+
+
+# This can be removed when support for python<3.8 is dropped
+def copytree_dirs_exist_ok(
+    src_top: str | pathlib.Path, dst_top: str | pathlib.Path
+) -> None:
+    src_top = pathlib.Path(src_top)
+    dst_top = pathlib.Path(dst_top)
+
+    dst_top.mkdir(exist_ok=True)
+    for src in src_top.iterdir():
+        dst = dst_top / src.name
+        if src.is_file():
+            shutil.copy2(src, dst)
+        else:
+            shutil.copytree(src, dst)
+
+
+@pytest.mark.network
+@pytest.mark.parametrize(
+    ("distributions", "expected_deps"),
+    (
+        (
+            ["editable"],
+            {
+                "setuptools": Version("*"),
```

Comment:
I think it would be nice to depend less, or not at all, on PyPI for these tests. If others agree then I will look a bit more at avoiding it and I'm happy for advice and suggestions.

One idea is to download some well defined distribution packages into `fake_dists` and to use `make_pip_conf` to set `no-index = true`.

Reply:
Yes, allowing tests to run offline is a good idea, but shouldn't necessarily be bundled with this PR.